sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -470,6 +470,11 @@ object SimplifyConditionals extends Rule[LogicalPlan] with PredicateHelper {
     case _ => false
   }
 
+  private def isAlwaysFalse(exps: Seq[Expression], equalTo: Literal): Boolean = {
+    exps.forall(_.isInstanceOf[Literal]) &&
+      exps.forall(!EqualTo(_, equalTo).eval(EmptyRow).asInstanceOf[Boolean])
```

Comment:
`EqualTo` may return null, we need to take care of it.

Reply:
Yes. Null value should be handled by [NullPropagation](https://github.com/apache/spark/blob/831bf66b889d72724c072f9b9a20d8a436d4dbb8/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala#L636-L639).